java/src/main/native/src/TableJni.cpp

```diff
@@ -935,6 +935,39 @@ jlongArray combine_join_results(JNIEnv *env, cudf::table &left_results,
   return combine_join_results(env, std::move(left_cols), std::move(right_cols));
 }
 
+cudf::column_view remove_validity_from_col(cudf::column_view column_view) {
+  if (column_view.nullable() && column_view.null_count() == 0) {
+    // null_mask is allocated but no nulls present therefore we create a new column_view without the
+    // null_mask to avoid things blowing up in reading the parquet file
+    if (!cudf::is_nested(column_view.type())) {
+      cudf::column_view cv(column_view.type(), column_view.size(), column_view.data<int32_t>(),
+                           nullptr, 0);
+      return cv;
+    } else {
+      std::vector<cudf::column_view> children;
+      children.reserve(column_view.num_children());
+      for (auto it = column_view.child_begin(); it != column_view.child_end(); it++) {
+        children.push_back(remove_validity_from_col(*it));
+      }
+      cudf::column_view cv(column_view.type(), column_view.size(), column_view.data<int32_t>(),
+                           nullptr, 0, 0, children);
```

Comment:
```suggestion
      cudf::column_view cv(column_view.type(), column_view.size(), nullptr,
                           nullptr, 0, column_view.offset(), children);
```
Parent columns have no data and the `data()` method accounts for `offset` so it would be bad to pass what it returns here.